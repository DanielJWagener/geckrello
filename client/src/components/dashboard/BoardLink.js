import React from "react";
import { Link } from "react-router-dom";

const BoardLink = props => {
  return (
    <Link to={`/board/${props.boardId}`}>
      <h1>{`I'm a link to ${props.title}`}</h1>;
    </Link>
  );
};

export default BoardLink;
