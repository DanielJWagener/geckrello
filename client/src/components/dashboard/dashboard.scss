.dashboard {
  //background-color: #eeeeee;
  // height: 100%;
}
