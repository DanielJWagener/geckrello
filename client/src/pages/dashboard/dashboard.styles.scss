.dashboard {
  background-color: #eeeeee;
  height: calc(var(--vh, 1vh) * 100 - var(--navbar-height));
  padding: 2rem;

  &__container {
    max-width: 1040px;
    margin-left: auto;
    margin-right: auto;
  }

  &__welcome-message {
    margin-bottom: 1rem;
  }

  &__header {
    font-size: 3rem;
    margin-bottom: 1rem;
  }

  &__boards-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: stretch;
  }

  @media only screen and (max-width: 30.75em) {
    padding: 2rem 1rem;
    &__boards-container {
      flex-direction: column;
    }
  }
}
