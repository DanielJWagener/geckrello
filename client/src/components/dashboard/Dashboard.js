import React from "react";
import { connect } from "react-redux";

import AddBoard from "./AddBoard";
import BoardLink from "./BoardLink";
import { fetchUser } from "../../actions";
import "./dashboard.scss";

class Dashboard extends React.Component {
  componentDidMount() {
    this.props.fetchUser();

    let root = document.documentElement;

    root.style.setProperty("--body-background", "#eeeeee");
    //root.style.setProperty("--body-background-light", props.colorLight);
  }

  renderContent(auth) {
    switch (auth) {
      case null:
        return <h1>Loading...</h1>;
      case false:
        return <h1>You are not logged in!</h1>;
      default:
        if (this.props.auth.givenName) {
          return (
            <>
              <h1>{`Welcome, ${this.props.auth.givenName}!`}</h1>
              {this.renderBoards()}
              <AddBoard />
            </>
          );
        } else {
          return <h1>Welcome!</h1>;
        }
    }
  }

  renderBoards = () => {
    return this.props.auth.boards.map(board => {
      return (
        <BoardLink key={board._id} title={board.title} boardId={board._id} />
      );
    });
  };

  render() {
    return (
      <div className="dashboard">{this.renderContent(this.props.auth)}</div>
    );
  }
}

function mapStateToProps({ auth, boards }) {
  return { auth, boards };
}

export default connect(mapStateToProps, { fetchUser })(Dashboard);
