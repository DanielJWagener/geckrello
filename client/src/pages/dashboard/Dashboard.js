import React from "react";
import { connect } from "react-redux";

import AddBoard from "../../components/AddBoard/add-board.component";
import BoardLink from "../../components/BoardLink/board-link.component";
import { fetchUser } from "../../actions";
import "./dashboard.scss";
import setDefaultColors from "../../utilities/setDefaultColors";

class Dashboard extends React.Component {
  componentDidMount() {
    this.props.fetchUser();

    document.title = `Dashboard - Geckrello, a Trello Clone`;
    setDefaultColors();
    let root = document.documentElement;

    root.style.setProperty("--color-primary", "#eeeeee");
  }

  renderContent(auth) {
    switch (auth) {
      case null:
        return <h1>Loading...</h1>;
      case false:
        return <h1>You are not logged in!</h1>;
      default:
        if (this.props.auth.givenName) {
          return (
            <>
              <h1>{`Welcome, ${this.props.auth.givenName}!`}</h1>
              {this.renderBoards()}
              <AddBoard />
            </>
          );
        } else {
          return <h1>Welcome!</h1>;
        }
    }
  }

  renderBoards = () => {
    return this.props.auth.boards.map(board => {
      return (
        <BoardLink key={board._id} title={board.title} boardId={board._id} />
      );
    });
  };

  render() {
    return (
      <div className="dashboard">{this.renderContent(this.props.auth)}</div>
    );
  }
}

function mapStateToProps({ auth, boards }) {
  return { auth, boards };
}

export default connect(mapStateToProps, { fetchUser })(Dashboard);
