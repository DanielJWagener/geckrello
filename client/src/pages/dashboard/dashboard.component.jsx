import React from "react";
import { connect } from "react-redux";

import AddBoard from "../../components/AddBoard/add-board.component";
import BoardLink from "../../components/BoardLink/board-link.component";
import { fetchUser } from "../../actions";
import setDefaultColors from "../../utilities/setDefaultColors";

import "./dashboard.styles.scss";

export class Dashboard extends React.Component {
  componentDidMount() {
    this.props.fetchUser();

    document.title = `Dashboard - Geckrello, a Trello Clone`;
    setDefaultColors();
  }

  renderContent(auth) {
    switch (auth) {
      case null:
        return <h1>Loading...</h1>;
      case false:
        return <h1>You are not logged in!</h1>;
      default:
        const welcomeMessage = this.props.auth.givenName
          ? `Welcome, ${this.props.auth.givenName}!`
          : "Welcome!";

        return (
          <>
            <h1>{welcomeMessage}</h1>
            <div className="dashboard__boards-container">
              <AddBoard />
              {this.renderBoards()}
            </div>
          </>
        );
    }
  }

  renderBoards = () => {
    return this.props.auth.boards.map(({ _id, ...otherProps }) => (
      <BoardLink key={_id} boardId={_id} {...otherProps} />
    ));
  };

  render() {
    return (
      <div className="dashboard">{this.renderContent(this.props.auth)}</div>
    );
  }
}

function mapStateToProps({ auth, boards }) {
  return { auth, boards };
}

export default connect(mapStateToProps, { fetchUser })(Dashboard);
