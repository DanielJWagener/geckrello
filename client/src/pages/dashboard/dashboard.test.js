import { shallow } from "enzyme";
import React from "react";

import { Dashboard as DashboardPure } from "./dashboard.component";
const mockProps = {
  auth: {
    givenName: "",
    boards: [],
  },
  fetchUser: () => {},
};

it("renders Dashboard component", () => {
  expect(shallow(<DashboardPure {...mockProps} />).debug()).toMatchSnapshot();
});
