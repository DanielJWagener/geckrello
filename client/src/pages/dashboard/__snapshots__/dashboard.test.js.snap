// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard component 1`] = `
"<div className=\\"dashboard\\">
  <h1>
    Welcome!
  </h1>
  <div className=\\"dashboard__boards-container\\">
    <Connect(AddBoard) />
  </div>
</div>"
`;
